code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -287,11 +287,45 @@
 	if((I.body_parts_covered & FEET) || (I.flags_inv | I.transparent_protection) & HIDESHOES)
 		SEND_SIGNAL(src, COMSIG_CARBON_UNEQUIP_SHOECOVER, I, force, newloc, no_move, invdrop, silent)
 
+/mob/living/carbon/human/toggle_internals(obj/item/tank, is_external = FALSE)
+	// Just close the tank if it's the one the mob already has open.
+	var/obj/item/existing_tank = is_external ? external : internal
+	if(tank == existing_tank)
+		return toggle_close_internals(is_external)
+	// Use breathing tube regardless of mask.
+	if(can_breathe_tube())
+		return toggle_open_internals(tank, is_external)
+	// Use mask in absence of tube.
+	if(isclothing(wear_mask) && ((wear_mask.visor_flags & MASKINTERNALS) || (wear_mask.clothing_flags & MASKINTERNALS)))
+		// Adjust dishevelled breathing mask back onto face.
+		if (wear_mask.mask_adjusted)
+			wear_mask.adjustmask(src)
+		return toggle_open_internals(tank, is_external)
+	// Use helmet in absence of tube or valid mask.
+	if(can_breathe_helmet())
+		return toggle_open_internals(tank, is_external)
+	// Invalid helmet and missing mask.
+	// Don't show the "isn't sealed" message for non-helmet headgear, such as hats.
+	if (head && (istype(head, /obj/item/clothing/head/mod) || istype(head, /obj/item/clothing/head/helmet)))
```

Comment:
i dont like this check, why not just check for headwear and put this after the mask check

Reply:
I thought this out quite a bit. I implemented it this way to decouple the chat notification from the actual usage of the helmet. Not all headwear makes sense with a "this isn't sealed" message, such as a flower on your head. With the check set up this way, it also adjusts the breath mask even if you have a helmet on, assuming that user wants to use their worn mask.

Reply:
Fixed by commit c439d7f25b5c7423a1b6a5c242af2656ce4044d6